test/requirements.in

```diff
@@ -17,4 +17,5 @@ black
 isort
 flake8
 httpx
-pydantic
\ No newline at end of file
+pydantic
+respx
```

Comment:
Adding `respx` because MagicMocks cannot be awaited in python 3.7. We could write the code to work around this ourselves but probably better to use the best practice for mocking httpx

https://lundberg.github.io/respx/

Reply:
This also helps us do #1814 while we're at it

Reply:
Cool